app/view_objects/results_view.rb

```diff
@@ -67,6 +67,53 @@ def number_of_extra_subjects
     number_of_subjects_selected - NUMBER_OF_SUBJECTS_DISPLAYED
   end
 
+  def location
+    query_parameters["loc"] || "Across England"
+  end
+
+  def distance
+    query_parameters["rad"]
+  end
+
+  def latitude
+    query_parameters["lat"]
+  end
+
+  def longitude
+    query_parameters["lng"]
+  end
+
+  def show_map?
+    latitude.present? && longitude.present? && distance.present?
+  end
+
+  def google_map_zoom
+    case distance
+    when "5"
+      "12"
+    when "10"
+      "11"
+    when "20"
+      "10"
+    when "50"
+      "9"
+    when "100"
+      "8"
+    else
+      "14"
+    end
+  end
+
+  def map_image_url
```

Comment:
i think these should be in private
`latitude` , `longitude`, `google_map_zoom`, 

`"https://maps.googleapis.com/maps/api/staticmap` this can probably go in `Settings.google.xxx` as well